### DISCUSSION TITLE
MPE support

### DISCUSSION THREAD

**Discussion by muziker** [FluidSynth/fluidsynth#853] [Ideas]
I'd like mpe support to be built into fluidsynth. With this, it becomes possible to use individual pitch bends on every channel as opposed to only one pitch bend per fluidsynth instance. This enables the use of microtonal scales. I'd like to be able to use an orchestral library and adopt a microtonal scale over it.

**Comment by derselbst** [comment]
> With this, it becomes possible to use individual pitch bends on every channel

Every MIDI channel has its own pitch bend by default.

> as opposed to only one pitch bend per fluidsynth instance

This is wrong.

> This enables the use of microtonal scales. 

Fluidsynth already supports MTS. But this has nothing to do with MPE. MPE was invented to allow controlling notes individually in terms of CC, pitch, aftertouch, etc.

**Reply by muziker** [reply]
I was a little terse on the microtonal part. But enable the use of microtonal scales by means of midi note+pitch bending on a per channel basis, it's a use case of mine.

**Reply by derselbst** [reply]
You should not abuse the pitch bend for microtonal tuning. MTS allows you to tune single notes. This _is_ what fluidsynth [can do](https://www.fluidsynth.org/api/group__tuning.html#ga3a9f995c967a9a227f9603ff57ee2304). You don't need MPE for this.

**Comment by muziker** [comment]
From the man page, it seems that specifying a tuning scale and storing it
to a program and issuing midi program changes should work? For a given
note, I'm trying to glissando to the target tuning though, which is why i'm
using the pitch bending technique.

**Reply by derselbst** [reply]
> From the man page, it seems that specifying a tuning scale and storing it to a program and issuing midi program changes should work?

The tuning program is not related to the midi program. And tunings must be explicitly activated. We have little about tuning, so you'll have to figure it out yourself.

> I'm trying to glissando to the target tuning though, which is why i'm using the pitch bending technique.

Sounds like portamento/legato would be a better solution than pure pitch bend. [See the polymono documentation](https://github.com/FluidSynth/fluidsynth/tree/master/doc/polymono) for details and examples.
